### PR TITLE
Detach interface match by alias

### PR DESCRIPTION
**Pull request by yalzhang**
When there are 2 interfaces with duplicate mac address, detach by
mac address will fail. So libvirt add one more match method to match
interface by alias. Add 2 test cases both positive and negative to
test the match method.

Signed-off-by: Yalan <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
automate RHEL-202435

**Comment by yalzhang** [comment]
New add 2 cases:
virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_alias_dup_mac
virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_dup_mac

all related cases run locally pass.
$ avocado run --vt-type libvirt virtual_network.iface_hotplug..detach_match_test
JOB ID     : b4e9af07e7e57d7c40ada71550d63065bad60c65
JOB LOG    : /root/avocado/job-results/job-2021-07-06T22.32-b4e9af0/job.log
 (1/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_pci: PASS (96.28 s)
 (2/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_mac_pci: PASS (82.17 s)
 (3/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_mac: PASS (86.54 s)
 (4/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_wrong_mac: PASS (71.46 s)
 (5/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_wrong_pci: PASS (56.01 s)
 (6/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_alias_dup_mac: PASS (84.19 s)
 (7/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_hotplug.at_device.iface_attach.model_virtio.detach_match_test.by_dup_mac: PASS (70.83 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 554.67 s

File: libvirt/tests/cfg/virtual_network/iface_hotplug.cfg [hunk 1]
```diff
@@ -122,10 +122,26 @@
...
+                            del_mac = "no"
+                            start_vm = "no"
+                            attach_option = "--persistent"
+                            iface_num = '2'
+                            iface_mac = "9a:9b:b9:f5:51:b0"
+                            del_alias = "yes"
+                            detach_error = "multiple devices matching MAC"
+                            status_error = "yes"
             variants:
-                - model_e1000:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virtual_network/iface_hotplug.cfg:126]: Is it intended to remove those cases?

Comment (yalzhang) [reply libvirt/tests/cfg/virtual_network/iface_hotplug.cfg:126]: Yes, e1000 is declared  to deprecate on rhel8, though the driver is still available on rhel 8 kernel. So I think no need to test it any more. In fact, it was added in the blacklist for a long time.

**Review by chunfuwen — APPROVED** [review]
LGTM

**Review by dzhengfy — APPROVED** [review]
LGTM
